clang/include/clang/Sema/Sema.h

```diff
@@ -10158,6 +10158,8 @@ class Sema final : public SemaBase {
   CompareReferenceRelationship(SourceLocation Loc, QualType T1, QualType T2,
                                ReferenceConversions *Conv = nullptr);
 
+  bool AllowMSLValueReferenceBinding(Qualifiers Quals, QualType QT);
```

Comment:
In case you don't implement my other suggestion, this can be static.
```suggestion
  static bool AllowMSLValueReferenceBinding(Qualifiers Quals, QualType QT);
```